server/Dockerfile

```diff
@@ -47,82 +50,68 @@ FROM "${DOCKER_HUB_PROXY}debian:bullseye-slim" as php-build
-        sed -i 's/-e //g' requirements.txt; pip3 wheel -r requirements.txt --no-cache-dir -w /wheels/
-
-    # install python stix
-    RUN git clone --depth 1 https://github.com/STIXProject/python-stix.git; \
-        cd python-stix || exit; python3 setup.py bdist_wheel -d /wheels; \
-        sed -i 's/-e //g' requirements.txt; pip3 wheel -r requirements.txt --no-cache-dir -w /wheels/
-
-    # install STIX2.0 library to support STIX 2.0 export:
-    # Original Requirements has a bunch of non-required pacakges, force it to only grab wheels for deps from setup.py
-    RUN git clone --depth 1 https://github.com/MISP/cti-python-stix2.git; \
-        cd cti-python-stix2 || exit; python3 setup.py bdist_wheel -d /wheels; \
-        echo "-e ." > requirements.txt; pip3 wheel -r requirements.txt --no-cache-dir -w /wheels/
-
-    # Install the new build tool
-    RUN pip3 install build
-
-    # install PyMISP
-    RUN git clone --depth 1 https://github.com/MISP/PyMISP.git; \
-        cd PyMISP || exit; python3 -m build --wheel --outdir /wheels
-
-    # install pydeep2 (drop-in replacement for pydeep)
-    RUN git clone --depth 1 https://github.com/JakubOnderka/pydeep.git; \
-        cd pydeep || exit; python3 setup.py bdist_wheel -d /wheels
-
-    # Grab other modules we need
-    RUN pip3 wheel --no-cache-dir -w /wheels/ plyara pyzmq redis python-magic lief>=0.13.1
-
-    # Remove extra packages due to incompatible requirements.txt files
-    WORKDIR /wheels
-    RUN find . -name "Sphinx*" | tee /dev/stderr | grep -v "Sphinx-1.5.5" | xargs rm -f
-
+    ARG MISP_TAG
+    ARG MISP_COMMIT
+    # Download MISP using git in the /var/www/ directory. Remove unnecessary items.
+    RUN <<-EOF
+            if [ ! -z "${MISP_COMMIT}" ]; then
+                git clone https://github.com/MISP/MISP.git /var/www/MISP && cd /var/www/MISP && git checkout "${MISP_COMMIT}"
+            else
+                git clone --branch "${MISP_TAG}" --depth 1 https://github.com/MISP/MISP.git /var/www/MISP
+            fi
+
+            cd /var/www/MISP || exit; git submodule update --init --recursive .
+
+            # Until MISP project specifies its required PyMISP version, we grab this from the submodule
+            pymispversion=$(head -n1 PyMISP/pymisp/__init__.py | awk '{print $3}' | tr -d \')
+
+            if [ -n "$pymispversion" ]; then
+                sed -i "s/pymisp$/pymisp==$pymispversion/" requirements.txt
+                cat requirements.txt
+            fi
+            # End of pymisp version fixing hack
+
+            # Remove modules only used during MISP build
+            set -- 'coveralls' 'codecov$' 'requests-mock' 'nose'
+            for mod in "$@"; do
+                sed "/${mod}/d" -i /var/www/MISP/requirements.txt
+            done;
+
+            # Add additional dependencies (container specific)
+            set -- 'redis' 'lief>=0.13.1' 'pydeep2' 'python-magic' 'misp-lib-stix2' 'maec' 'mixbox'
```

Comment:
If the requirements file has lief>=0.14.1, lief>=0.13.1 will be added.
Can we define a mapping here with our pinnings, overriding those specified in requirements.txt ?

Reply:
Also, cybox is an explicit dependency that I see missing here